### PR TITLE
Add interoperation features

### PR DESCRIPTION
**Pull request by adekbadek**
### All Submissions:

* [x] Have you followed the [Newspack Contributing guideline](https://github.com/Automattic/newspack-plugin/blob/master/.github/CONTRIBUTING.md)?
* [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) and [VIP Go coding standards](https://vip.wordpress.com/documentation/vip-go/code-review-blockers-warnings-notices/)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](../../pulls) for the same update/change?

### Changes proposed in this Pull Request:

For a better interop with other plugins, two features are introduced here:

1. A hook which allows other plugins to receive the submitted email if a newsletter signup form in a Campaign was submitted
3. Blocks rendered within a Campaign are furnished with a `newspack-popups-content-block` class, for better event handling

### How to test the changes in this Pull Request:

1. Create a Campaign with a Newsletter signup block (Jetpack Mailchimp block)
2. In a different plugin, subscribe to `newspack_popups_mailing_list_subscription` hook
3. Visit an AMP page ([won't work with non-AMP](https://github.com/Automattic/newspack-popups/issues/200))
4. Subscribe to the newsletter 
5. Observe the hook receives a payload object with an `email` key, the value of which is the submitted email
6. Observe the blocks rendered within the Campaign have the `newspack-popups-content-block` class

### Other information:

* [x] Have you added an explanation of what your changes do and why you'd like us to include them?
* [ ] Have you written new tests for your changes, as applicable?
* [x] Have you successfully ran tests with your changes locally?

### CODE REVIEW

File: includes/class-newspack-popups-model.php [hunk 1]
```diff
@@ -591,6 +592,20 @@ protected static function get_analytics_events( $popup, $body, $element_id ) {
...
 
+	/**
+	 * Add "newspack-popups-content-block" class name to a block.
+	 * This way a block rendered inside of a popup can be easily told apart.
+	 *
+	 * @param object $block A block.
+	 * @return object Block with className appended.
+	 */
+	public static function append_class_to_block( $block ) {
+		if ( isset( $block['attrs']['className'] ) ) {
```
[the remark this answers is not in the document]
Comment (adekbadek) [reply includes/class-newspack-popups-model.php:604]: My bad, fixed in f815727fe98c4863c5be2c2d8ed2ca713f1a3ebb

File: includes/class-newspack-popups-model.php [hunk 2]
```diff
@@ -591,6 +592,24 @@ protected static function get_analytics_events( $popup, $body, $element_id ) {
...
 
+	/**
+	 * Add "newspack-popups-content-block" class name to a block.
+	 * This way a block rendered inside of a popup can be easily told apart.
+	 * This will not work which have HTML content assigned in the editor (most basic blocks)
+	 *
+	 * @param object $block A block.
+	 * @return object Block with className appended.
+	 */
+	public static function append_class_to_block( $block ) {
```
[the remark this answers is not in the document]
Comment (adekbadek) [reply includes/class-newspack-popups-model.php:603]: Good catch, thanks! Fixed in 75cf69933efca4232be71ec441686dfc2ec6b7ae

**Comment by adekbadek** [comment]
@jeffersonrabb 

> This seems like a reasonable temporary fix for capturing the email address after signup, but I think we will eventually need a more robust approach […] This could be rewritten where Jetpack has it's own endpoint which is simply a wrapper for a PHP call to the public API.

Great idea, I'll look into it

File: includes/class-newspack-popups-api.php [hunk 3]
```diff
@@ -362,6 +371,21 @@ public function get_mailing_list_status( $request ) {
...
+	 * Get email address.
+	 *
+	 * @param WP_REST_Request $request amp-access request.
+	 * @return string Email address.
+	 */
+	public function get_email_address( $request ) {
+		$email_address = isset( $request['email'] ) ? esc_attr( $request['email'] ) : false;
+		if ( ! $email_address ) {
+			$body          = json_decode( $request->get_body(), true );
+			$email_address = isset( $body['email'] ) ? $body['email'] : false;
```
Comment (jeffersonrabb) [inline includes/class-newspack-popups-api.php:384]: Shouldn't `$body['email']` be escaped like `$request['email']`?

Comment (adekbadek) [reply includes/class-newspack-popups-api.php]: Fixed in 7d6870866987d3356ccb19bfadd47bec4cd954a0

**Pull request merged**